db/helpers/botUtils.js

```diff
@@ -45,16 +45,23 @@ function arrayHasValidParameters(arr) {
   if (Array.isArray(arr)) {
     for (let i = 0; i < arr.length; i++) {
       if ((typeof arr[i] === 'object') &&
-        (arr[i].hasOwnProperty('name')) && (arr[i].hasOwnProperty('type'))) {
-        if ((!constants.nameRegex.test(arr[i].name)) ||
+        (arr[i].hasOwnProperty('name')) &&
```

Comment:
Can you take a look at https://www.npmjs.com/package/joi? Might be cleaner way to validate these objects.